tests/unit/machine/glm_unittest.cc

```diff
@@ -0,0 +1,115 @@
+/*
+ * This software is distributed under BSD 3-clause license (see LICENSE file).
+ *
+ * Both the documentation and the code is heavily inspired by pyGLMnet.:
+ * https://github.com/glm-tools/pyglmnet/
+ *
+ * Author: Tej Sukhatme
+ */
+
+#include <gtest/gtest.h>
+#include <shogun/features/DenseFeatures.h>
+#include <shogun/labels/RegressionLabels.h>
+#include <shogun/lib/SGMatrix.h>
+#include <shogun/machine/GLM.h>
+#include <shogun/machine/LinearMachine.h>
+#include <shogun/mathematics/RandomNamespace.h>
+#include <shogun/mathematics/UniformRealDistribution.h>
+#include <shogun/mathematics/eigen3.h>
+#include <shogun/mathematics/linalg/LinalgNamespace.h>
+#include <shogun/preprocessor/NormOne.h>
+#include <shogun/preprocessor/PruneVarSubMean.h>
+
+#include <random>
+
+using namespace shogun;
+
+// Generate the Data that N is greater than D
+std::tuple<SGMatrix<float64_t>, SGVector<float64_t>> generate_train_data()
+{
+	SGVector<float64_t> labels = SGVector<float64_t>(
+	    {7.23514031, 7.23514031, 7.23514031, 7.23514031, 7.23514031});
```

Comment:
why are all labels the same?

Reply:
This happened as the beta is supposed to be a sparse vector and so this happens I guess:
```
import numpy as np
import scipy.sparse as sps

from pyglmnet import simulate_glm

# sample a sparse model
np.random.seed(42)
beta0 = np.random.rand()
beta = sps.random(1, n_features, density=0.2).toarray()[0]

# simulate data
Xtrain = np.random.normal(0.0, 1.0, [n_samples, n_features])
ytrain = simulate_glm('poisson', beta0, beta, Xtrain)
Xtest = np.random.normal(0.0, 1.0, [n_samples, n_features])
ytest = simulate_glm('poisson', beta0, beta, Xtest)
```

Reply:
`simulate_glm` misses the argument `sample=True`. You need that in order to simulate correctly the y labels.

Reply:
Oh okay, should I go ahead and change the unit test data?
Or implement this in the meta example test?

Reply:
Because this test doesn't exactly check if the GLM works properly, it checks if the GLM gives the exact same answer as the pyGLMnet code.
As in, the numbers don't matter as much right?

Reply:
> Oh okay, should I go ahead and change the unit test data?
> Or implement this in the meta example test?

I think you should change it such to have a meaningful unit test. It is true that they need to give the same results, but we have to try to use meaningful data.